### ISSUE TITLE
`abi-to-sol` fails to produce prettified output without being installed first

### ISSUE THREAD

**Issue by gosuto-inzasheru** [gnidan/abi-to-sol#67]
running the example from the readme i get this:
```
// SPDX-License-Identifier: UNLICENSED
// !! THIS FILE WAS AUTOGENERATED BY abi-to-sol v0.5.3. SEE SOURCE BELOW. !!
pragma solidity >=0.7.0 <0.9.0;

interface ENS {


function resolver( bytes32 node ) external view returns (address ) ;
function owner( bytes32 node ) external view returns (address ) ;
function setSubnodeOwner( bytes32 node,bytes32 label,address owner ) external   ;
function setTTL( bytes32 node,uint64 ttl ) external   ;
function ttl( bytes32 node ) external view returns (uint64 ) ;
function setResolver( bytes32 node,address resolver ) external   ;
function setOwner( bytes32 node,address owner ) external   ;
event Transfer( bytes32 indexed node,address owner ) ;
event NewOwner( bytes32 indexed node,bytes32 indexed label,address owner ) ;
event NewResolver( bytes32 indexed node,address resolver ) ;
event NewTTL( bytes32 indexed node,uint64 ttl ) ;
}
```
how can we make the indentations explicit? my compiler refuses an interface in this flat format ☹️

**Comment by gnidan** [comment]
Hey @jorijnsmit, sorry I missed the notification for this issue!

The CLI should run prettier automatically (looks like I didn't even make it configurable via flag, oops!). The setup for it is pretty brittle, though, so I programmed abi-to-sol to silently ignore any prettier errors and just output unprettified code.

I've got a team retreat next week so I may or may not have time to track down the bug until after that, but feel free to take a look yourself... if you can figure it out, you can be the first to join the robots and me on this project's contributors list :D

**Comment by jorijnsmit** [comment]
aah the issue was actually that i just didnt have the reqs installed.
```
npm install --save-dev prettier prettier-plugin-solidity
```
did the trick :)

**Comment by jorijnsmit** [comment]
unfortunately im not at home in npm package management, but is there a way to auto include these requirements somehow..?

**Comment by gnidan** [comment]
> is there a way to auto include these requirements somehow..?

Hmm... I listed them as `optionalDependencies` [here](https://github.com/gnidan/abi-to-sol/blob/develop/packages/abi-to-sol/package.json#L58,L61), and looks like NPM does try to install these; it just allows the rest of the process to continue (according to the [documentation for that field](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#optionaldependencies)).

So sounds like when you installed abi-to-sol, prettier and prettier-plugin-solidity must have errored for some reason. Maybe there was console output showing error information when you installed?

I'll have to try to reproduce this. What Node/NPM version are you on? And are you doing `npx abi-to-sol` or `npm install abi-to-sol`?

**Comment by gnidan** [comment]
> aah the issue was actually that i just didnt have the reqs installed.
> 
> ```
> 
> npm install --save-dev prettier prettier-plugin-solidity
> 
> ```
> 
> did the trick :)

Anyway, glad to hear there's a simple workaround in the meantime!

**Comment by gnidan** [comment]
Reopening this since I think there's likely more that remains to be done here!

**Comment by esaulpaugh** [comment]
In the web tool https://gnidan.github.io/abi-to-sol/ I notice that if I specify "type" as "function" in one of the "inputs" inside an event or function, or as one of the "outputs" inside a function, the output will generate, but prettify will automatically be disabled.

**Comment by gnidan** [comment]
> In the web tool https://gnidan.github.io/abi-to-sol/ I notice that if I specify "type" as "function" in one of the "inputs" of an event or function, or as one of the "outputs" of a function, the output will generate, but prettify will automatically be disabled.

Hm, I wonder if this is a bug with prettier-plugin-solidity. Does the non-pretty output still compile correctly?

**Comment by esaulpaugh** [comment]
The generated output doesn't seem to compile:

```solidity
pragma solidity >=0.7.0 <0.9.0;

interface MyInterface {


function zzz( function aaa ) external pure  ;
}
```

**Comment by gnidan** [comment]
I'll double-check this later today or tomorrow. You may have found an abi-to-sol bug! Thank you!

**Comment by gnidan** [comment]
@esaulpaugh I just tested this... you're absolutely right, this is something that doesn't work right! I opened a fresh issue for it, as you can see linked above ^

Thank you!

**Comment by gnidan** [comment]
Getting back to the original issue, I just released [abi-to-sol v0.6.4](https://github.com/gnidan/abi-to-sol/releases/tag/v0.6.4) with upgrades to prettier and prettier-plugin-solidity, hoping this would fix the issue, but it seems that `npx abi-to-sol` still fails without a prior `npm install`.

It's probably not worth maintaining the requirement to support a fresh `npx abi-to-sol`, since I wrote the web UI for getting started quickly, and arguably that's even easier than going to a terminal and `cat`-ing stuff around. Instead, I've just removed this from the README. Closing this issue now as won't-fix.

Thanks again for bringing this up @gosuto-inzasheru!
